### PR TITLE
Feature/11 list

### PR DESCRIPTION
**Pull request by ostatni5**
Sample results list 

- filter results including string
- sort by column
- hide column 

obraz

### CODE REVIEW

File: ui/src/lib/resultsList/ResultsList.svelte [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<script lang="ts">
+	import SvelteTable from 'svelte-table/src/index';
```
Comment (michal367) [inline ui/src/lib/resultsList/ResultsList.svelte:2]:
```suggestion
	import SvelteTable from 'svelte-table';
```

File: ui/src/lib/resultsList/ResultsList.svelte [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		prepareColumns['url'].renderValue = (v) =>
+			`<img src="https://picsum.photos/seed/${v.name}/20/20" alt="icon" width=20 height=20 >`;
+		prepareColumns['url'].title = 'IMAGE';
+
+		for (const key in prepareColumns) {
+			columns.push(prepareColumns[key]);
+		}
+	}
+
+	const exludeColumns = ['url'];
```
Comment (michal367) [inline ui/src/lib/resultsList/ResultsList.svelte:31]:
```suggestion
	const excludeColumns = ['url'];
```

File: ui/src/lib/resultsList/ResultsList.svelte [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+				}
+				return false;
+		  })
+		: rows;
+
+	let hiddenColumns = [];
+	$: visibleColumns = columns.filter((el) => !hiddenColumns.includes(el.key));
+</script>
+
+<label>Search in resluts: <input bind:value={search} /></label>
```
Comment (michal367) [inline ui/src/lib/resultsList/ResultsList.svelte:52]:
```suggestion
<label>Search in results: <input bind:value={search} /></label>
```

File: ui/src/lib/resultsList/ResultsList.svelte [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+<div>
+	<p>Hide columns</p>
```
Comment (michal367) [inline ui/src/lib/resultsList/ResultsList.svelte:54]: I think if we limit the number of colums to minimum then hiding columns will not be needed.
Also selecting checkbox to check which columns to hide isn't UX friendly.

**Review by michal367 — CHANGES_REQUESTED** [review]
obraz

I propose creating component with big image and under image its description instead of table rows (this is only a proposition, but something must be done with these small images).

And this description doesn't need to have all properties of this image, because if somebody searches for something all properties that he searches for will be visible for him in form above results.

Description could contain some basic info like: dimensions (with unit - px), size (with unit), name, path. Not every actual info in columns (like gender) is necessery.

File: ui/src/lib/resultsList/ResultsList.svelte (hunk 4 above)
Comment (ostatni5) [reply ui/src/lib/resultsList/ResultsList.svelte:54]: obraz
This is in task

File: ui/src/lib/resultsList/ResultsList.svelte (hunk 1 above)
Comment (ostatni5) [reply ui/src/lib/resultsList/ResultsList.svelte:2]: If you import like that JEST cant import it

**Comment by ostatni5** [comment]
> And this description doesn't need to have all properties of this image, because if somebody searches for something all properties that he searches for will be visible for him in form above results.

It is quite different from what I have done. Create for this another task.

**Comment by ostatni5** [comment]
obraz
Added bigger images

**Review by michal367 — APPROVED** [review]
I can accept this PR for now. But this thing still needs to be improved: e.g. styles, data display and data description etc.

**Pull request merged**